### PR TITLE
remove to read config yaml

### PR DESCRIPTION
**Pull request by simcod**
References #80

### CODE REVIEW

**Review by vknabel — COMMENTED** [review]
Great idea to remove viper altogether. Can you also remove it from the `go.mod` and run `go mod tidy`?

**Comment by vknabel** [comment]
It seems like there is still a usage of viper left in the code. That way a `go mod tidy` always re-adds it.
If I remember correctly the left occurrence could be replaced with an env access, too.

File: internal/provider/provider.go [hunk 1]
```diff
@@ -152,7 +150,7 @@ func (p *MetalstackCloudProvider) Configure(ctx context.Context, req provider.Co
 		BaseURL:   apiUrl,
 		Token:     apiToken,
 		UserAgent: "terraform-provider-metal/" + p.version,
-		Debug:     viper.GetBool("debug"),
+		Debug:     true,
```
Comment (simcod) [inline internal/provider/provider.go:153]: @vknabel Do you have an idea, how to replace this?

Comment (vknabel) [reply internal/provider/provider.go:153]: In `main.go` there is a local `debug` variable which you could make global.

File: internal/shared-package/shared.go
Comment (vknabel) [inline internal/shared-package/shared.go:1]: Was there a reason why you didn't name the folder just `shared`?

Comment (simcod) [reply internal/shared-package/shared.go:1]: No, but I can do so:)

**Review by vknabel — APPROVED** [review]

**Pull request merged**